### DISCUSSION TITLE
Online assemble fail

### DISCUSSION THREAD

**Discussion by gitthangbaby** [007revad/Synology_HDD_db#295] [General]
What are the steps once "Online Assemble" fails?

I don't want to recreate the pool, as this can happen again, i want to fix it. What could be done if NVME (exists along with HDD) is recognized, but can't be assembled repeatedly?
"A drive іп this storage pool was once missing, but a sufficient number of drives has been detected."

I've read tickets and this is rarely discussed. I've ran commands and all looks good. I also removed patches, or used --restore or vice versa.

NAS synoscgi_SYNO.Storage.CGI.Pool_1_reassemble[19894]: feasibility_check.cpp:129 FeasibilityCheck: [Info] [0] of feasibility check [pool_online_assemble] failed.

**Comment by 007revad** [comment]
I'm not sure what exactly DSM's online assemble does or checks for.

This page, https://kb.synology.com/en-my/DSM/help/DSM/StorageManager/storage_pool_online_assemble?version=7 says:
> If the assembling fails, the status of the Available Pool will change to Assemble failed and the Online Assemble button will not be clickable. If this situation occurs, we recommend restarting your system to assemble the Available Pool.

I assume you've rebooted and tried again.

The bigger issue is why does it need assembling. And why does the feasibility check fail.

How did you originally create the storage pool?

**Comment by gitthangbaby** [comment]
The knowledge base doesn't say much. Of course i've restarted or updated HD db, or removed boot scripts, or reinstalled. The assembly check validates the compatibility, but that should be fine. So I think there was a corruption during forced shutdown. Probably an issue with Synology "root block" or "synoblock". This level of new abstraction is alarming, it will deem the future repairs very hard. I've noticed so many new wrappers and abstractions in DSM7. Not happy about that, as DSM6 repairs of botched pools was so simple. Now I know in advance i'm done once the problem reappears. That's why i'm trying to collect some information. Perhaps from folks who went through recovery.

I've created NVME and HDD storage pool. Apps are on NVME so it renders the system empty after such incident. With general lack of information, i gave up and recreated the pool. Recovery took me 2 days. Backups are weak as expected. Unlike Qnap, there's no Apps or System shared folder, so the admin must do their own root rsync backups. Which I do, but it still takes so much time to recover all the settings and files.

* HyperBackup is so silly that it forces inclusion of shares in application backup. So i can't turn it on for some important apps. You need to review every single setting afterwards, reconfigure permissions of each share, recreate snapshot settings etc.
* Replication is so poor, there's yet another level of abstraction, and you cannot recover from target drive in my case, because the link is broken. What a joke, a useless replication. I did use btrfs send|receive, but couldn't reconnect to the copies, so i've had to replicate it back to the target with the same data.  Insane.

So imagine, having 3 backups for the same data, and yet the recovery is a painful manual procedure.
That's why any ideas are welcome. I'm sure the problem will reappear.

**Reply by 007revad** [reply]
It does sound like there was some corruption in the partitioning on the drive. I know how to create the syno partitions but it deletes all the data on the drive. 

When I've used hyperbackup to backup the few packages that hyperbackup supports I select the smallest shared folder I have.

I use my own script to backup packages to another volume. https://github.com/007revad/Synology_app_mover When I select restore it restores the settings and data for the package, even if the package is now installed on a different volume to when it was backed up.

Repairing DSM 6 style storage pools was easy. DSM 7 is more complicated, and that's without considering encrypted volumes.

**Reply by 007revad** [reply]
Which model Synology do you have?

**Reply by gitthangbaby** [reply]
> Which model Synology do you have?

SA6400, Arc custom kernel, nvmesystem module., running in VM but disks are attached physically via PCI. This model has the best support for NVME, can accept NVME only.
